server/policies/collection.ts

```diff
@@ -72,89 +75,76 @@ allow(User, "share", Collection, (user, collection) => {
   if (!collection.sharing) {
     return false;
   }
-  if (user.isAdmin) {
-    return true;
-  }
 
   if (
     collection.permission !== CollectionPermission.ReadWrite ||
     user.isViewer
   ) {
-    invariant(
-      collection.memberships,
-      "membership should be preloaded, did you forget withMembership scope?"
-    );
-    const allMemberships = [
-      ...collection.memberships,
-      ...collection.collectionGroupMemberships,
-    ];
-    return some(
-      allMemberships,
-      (m) => m.permission === CollectionPermission.ReadWrite
-    );
+    return includesMembership(collection, [
+      CollectionPermission.ReadWrite,
+      CollectionPermission.Admin,
+    ]);
   }
 
   return true;
 });
 
-allow(User, ["publish", "update"], Collection, (user, collection) => {
+allow(User, "readDocument", Collection, (user, collection) => {
   if (!collection || user.teamId !== collection.teamId) {
     return false;
   }
-  if (user.isAdmin) {
-    return true;
-  }
 
-  if (
-    collection.permission !== CollectionPermission.ReadWrite ||
-    user.isViewer
-  ) {
-    invariant(
-      collection.memberships,
-      "membership should be preloaded, did you forget withMembership scope?"
-    );
-    const allMemberships = [
-      ...collection.memberships,
-      ...collection.collectionGroupMemberships,
-    ];
-    return some(
-      allMemberships,
-      (m) => m.permission === CollectionPermission.ReadWrite
-    );
+  if (collection.isPrivate) {
+    return includesMembership(collection, Object.values(CollectionPermission));
   }
 
   return true;
 });
 
-allow(User, "delete", Collection, (user, collection) => {
+allow(
+  User,
+  ["updateDocument", "createDocument", "deleteDocument"],
```

Comment:
Note: All these new policies, separated from "read", "update" of collection itself.